jobsdb/jobsdb.go

```diff
@@ -1176,18 +1190,35 @@ func (jd *HandleT) checkIfMigrateDS(ds dataSetT) (bool, int) {
 	err = row.Scan(&minCreatedAt, &maxCreatedAt)
 	jd.assertError(err)
 
-	if jd.MinDSRetentionPeriod > time.Duration(0) && time.Since(maxCreatedAt) < jd.MinDSRetentionPeriod {
-		return false, totalCount - delCount
+	jd.dsMigrationLock.RLock()
+	retention, ok := jd.dsRetentionMap[ds.JobTable]
+	jd.logger.Debugf(
+		"MinDSRetentionPeriod: %s, retention: %s, ok: %v, ds.JobTable: %s, dsRetentionMap: %+v",
+		jd.MinDSRetentionPeriod, time.Since(retention), ok, ds.JobTable, jd.dsRetentionMap,
+	)
+	jd.dsMigrationLock.RUnlock()
+	if jd.MinDSRetentionPeriod > 0 {
+		if time.Since(maxCreatedAt) < jd.MinDSRetentionPeriod ||
+			(ok && time.Since(retention) < jd.MinDSRetentionPeriod) {
+			return false, false, recordsLeft
+		}
 	}
-	if jd.MaxDSRetentionPeriod > time.Duration(0) && time.Since(minCreatedAt) > jd.MaxDSRetentionPeriod {
-		return true, totalCount - delCount
+
+	if jd.MaxDSRetentionPeriod > 0 && time.Since(minCreatedAt) > jd.MaxDSRetentionPeriod {
+		return true, false, recordsLeft
 	}
 
 	if (float64(delCount)/float64(totalCount) > jobDoneMigrateThres) ||
 		(float64(statusCount)/float64(totalCount) > jobStatusMigrateThres) {
-		return true, totalCount - delCount
+		return true, false, recordsLeft
```

Comment:
We should be honest here about whether this is a small table or not. 
As mentioned earlier, we wouldn't want to migrate this table by itself if it is small, even though the `jobDoneMigrateThres` or `jobStatusMigrateThres` have been surpassed.

Please let me provide some examples that we would like to avoid:

Assuming that 
- jobDoneMigrateThres: 0.8
- jobStatusMigrateThres: 5

1. We don't want to migrate a small table alone with 9/10 jobs completed (0.9 > 0.8)
2. We don't want to migrate a small table alone with 10 failed jobs for which we have written 60 job statuses (6>5)